gnuradio-runtime/lib/logger.cc

```diff
@@ -1,416 +1,157 @@
-                                 std::string target,
-                                 std::string pattern)
+logger::logger(const std::string& logger_name)
+    : _name(logger_name),
+      d_logger(
+          std::make_shared<spdlog::logger>(_name, logging::singleton().default_backend()))
 {
-    logger->removeAllAppenders();
-    logger_add_console_appender(logger, target, pattern);
+    d_logger->set_level(logging::singleton().default_level());
+    // gr::log :%p: %c{1} - %m%n
+    //         :level: block alias - message
+    d_logger->set_pattern(fmt::format(logging::default_pattern, _name));
 }
 
-void logger_add_file_appender(logger_ptr logger,
-                              std::string filename,
-                              bool append,
-                              std::string pattern)
-{
-    log4cpp::PatternLayout* layout = new log4cpp::PatternLayout();
-    log4cpp::Appender* app =
-        new log4cpp::FileAppender("FileAppender::" + filename, filename, append);
-    layout->setConversionPattern(pattern);
-    app->setLayout(layout);
-    logger->setAppender(app);
-}
+log_level logger::get_level() const { return d_logger->level(); }
 
-void logger_set_file_appender(logger_ptr logger,
-                              std::string filename,
-                              bool append,
-                              std::string pattern)
+const std::string logger::get_string_level() const
 {
-    logger->removeAllAppenders();
-    logger_add_file_appender(logger, filename, append, pattern);
+    const auto view = spdlog::level::to_string_view(d_logger->level());
+    return std::string(view.begin(), view.end());
 }
 
-void logger_add_rollingfile_appender(logger_ptr logger,
-                                     std::string filename,
-                                     size_t filesize,
-                                     int bkup_index,
-                                     bool append,
-                                     mode_t mode,
-                                     std::string pattern)
-{
-    log4cpp::PatternLayout* layout = new log4cpp::PatternLayout();
-    log4cpp::Appender* app = new log4cpp::RollingFileAppender(
-        "RollFileAppender::" + filename, filename, filesize, bkup_index, append, mode);
-    layout->setConversionPattern(pattern);
-    app->setLayout(layout);
-    logger->setAppender(app);
-}
+void logger::get_level(std::string& level) const { level = get_string_level(); }
+void logger::set_level(const log_level level) { d_logger->set_level(level); }
 
-std::vector<std::string> logger_get_logger_names(void)
+void logger::set_level(const std::string& level)
 {
-    std::vector<std::string> names;
-    std::vector<log4cpp::Category*>* loggers = log4cpp::Category::getCurrentCategories();
-    std::vector<log4cpp::Category*>::iterator logger = loggers->begin();
-
-    for (; logger != loggers->end(); logger++) {
-        names.push_back((*logger)->getName());
+    auto l = spdlog::level::from_str(level);
```

Comment:
Can you please not do the single-character vars? Here, `l` is level, below, it's `default_logger` (or something like that).